src/frontend/js/hooks/useUnionResource/index.ts

```diff
@@ -90,6 +90,8 @@ const useUnionResource = <
   const eofRef = useRef<Record<string, number>>(queryClient.getQueryData(eofQueryKey) ?? {});
   log('eof', eofRef.current);
 
+  const [unionQueryKey, setUnionQueryKey] = useState<string>();
```

Comment:
You should populate the `useState` with the current filters to prevent to trigger a reset on mount.